sentenai/flare.py

```diff
@@ -319,80 +498,179 @@ def __str__(self):
+
+        Arguments:
+            sw -- TODO: define this.
+        """
         if sw is None:
             b = {'name': self._name}
             if self._filters:
                 if len(self._filters) > 1:
-                    b['filter'] = {'type': '&&', 'args': [x() for x in self._filters]}
+                    b['filter'] = {
+                        'type': '&&',
+                        'args': [x() for x in self._filters]
+                    }
                 elif len(self._filters) == 1:
                     b['filter'] = self._filters[0]()
             return b
         else:
             try:
                 return sw._bind(self)
             except AttributeError as e:
-                raise TypeError("A stream should not be called with " + str(type(sw)), e)
+                raise TypeError(
+                    "A stream should not be called with " + str(type(sw)), e)
 
     def __getattr__(self, name):
+        """Get a SteamPath for a stream.
+
+        StreamPaths are used to reference variables in events themselves.
+
+        >>> s = stream("foo")
+        >>> s.foo.bar.bat.baz
+
+        Arguments:
+            name -- The name of the variable to get
+        """
         return StreamPath((name,), self)
 
     def _(self, name):
+        """Get a StreamPath for a stream.
+
+        Used primarily to escape segments of paths that would be invalid
+        in the host language. For example, if a path segment contains `:`
+
+        >>> s = stream("foo")
+        >>> s.bar._("...").baz.bat
+
+        Arguments:
+            name -- The name of the variable to get.
+        """
         return StreamPath((name,), self)
 
 
 @py2str
 class EventPath(object):
+    """An event's attribute path.
+
+    Used to reference variables within a single event. Combine with operators
+    to create condition objects.
+    """
+
     def __init__(self, namet=None):
-        if namet == None:
+        """Initialize the event path.
+
+        Arguments:
+            namet -- A list of variable names used to costruct a path. E.g.
+                     ['foo', 'bar', 'baz'] becomes 'foo.bar.baz'
+        """
+        if not namet:
             self.__attrlist = tuple()
         else:
             self.__attrlist = tuple(namet)
 
     def __getattr__(self, name):
+        """Get an EventPath for an event.
+
+        Used to reference variables within an event.
+
+        >>> evt.foo.bar.bat
+
+        Arguments:
+            name -- the name of the variable to get.
+        """
         return EventPath(self.__attrlist + (name,))
 
     def _(self, name):
+        """Get an EventPath for an event.
+
+        Used primarily to escape segments of paths that would be invalid
+        in the host language. For example, if a path segment contains `:
+
+        >>> evt.foo._('...').bar.bat
+
+        Arguments:
+            name -- the name of the variable to get.
+        """
+<<<<<<< HEAD
```

Comment:
Looks like the merge conflict wasn't resolved properly.

Reply:
Oops. Forgot to save changes to this file. Should be fixed now.